jlm/llvm/opt/alias-analyses/Andersen.cpp

```diff
@@ -899,7 +899,7 @@ Andersen::Analyze(const RvsdgModule & module)
 std::unique_ptr<PointsToGraph>
 Andersen::ConstructPointsToGraphFromPointerObjectSet(
     const PointerObjectSet & set,
-    Statistics & statistics)
+    Statistics * statistics)
```

Comment:
Why did you change this from a reference to a pointer?

Reply:
Is it because of the unit test? Could you not just create one there?

Reply:
That is what I started doing, but I realized the Statistics class is only forward declared in the header, while the definition is in `Andersen.cpp`.   Without moving it, or declaring a public `Andersen:: CreateStatistics()`, I can't make instances of Statistics from the test.

I don't know if it is possible to forward declare `Andersen::Statistics::Create()`, but googling doesn't give me any examples of it being allowed.

Reply:
Could you provide a `ConstructPointsToGraphFromPointerObjectSet()` method that does not take a `Statistics` and internally is defined in the `cpp` file and calls the method version with `Statistics` parameter internally?

Reply:
Good point, that works